src/toil_scripts/tools/aligners.py

```diff
@@ -56,3 +56,85 @@ def run_star(job, cores, r1_id, r2_id, star_index_url):
     return transcriptome_id, sorted_id
+
+
+def run_bwakit(job, config, threads, sort=True, trim=False):
+    """
+    Runs BWA-Kit to align a fastq file or fastq pair into a BAM file.
+
+    :param JobFunctionWrappingJob job: Passed by Toil automatically
+    :param Namespace config: A configuration object that holds strings as attributes.
+        The attributes must be accessible via the dot operator.
+        The config must have:
+        config.r1               FileStoreID for 1st fastq file
+        config.r2               FileStoreID for 2nd fastq file (or None if single-ended)
+        config.ref              FileStoreID for the reference genome
+        config.fai              FileStoreID for the reference index file
+        config.amb              FileStoreID for the reference amb file
+        config.ann              FileStoreID for the reference ann file
+        config.bwt              FileStoreID for the reference bwt file
+        config.pac              FileStoreID for the reference pac file
+        config.sa               FileStoreID for the reference sa file
+        config.alt              FileStoreID for the reference alt (or None)
+        config.rg_line          The read group value to use (or None -- see below)
+        config.library          Read group attribute: library
+        config.platform         Read group attribute: platform
+        config.program_unit     Read group attribute: program unit
+        config.uuid             Read group attribute: sample ID
+
+        If specifying config.rg_line, use the following format:
+            BAM read group header line (@RG), as defined on page 3 of the SAM spec.
+            Tabs should be escaped, e.g., @RG\\tID:foo\\tLB:bar...
+            for the read group "foo" from sequencing library "bar".
+            Multiple @RG lines can be defined, but should be split by an escaped newline \\n,
+            e.g., @RG\\tID:foo\\t:LB:bar\\n@RG\\tID:santa\\tLB:cruz.
+
+    :param int threads: Number of threads to use
+    :param bool sort: If True, sorts the BAM
+    :param bool trim: If True, performs adapter trimming
+    :return: FileStoreID of BAM
+    :rtype: str
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    file_names = ['r1.fq.gz', 'ref.fa.fai', 'ref.fa', 'ref.fa.amb', 'ref.fa.ann',
+                  'ref.fa.bwt', 'ref.fa.pac', 'ref.fa.sa']
+    ids = [config.r1, config.ref, config.fai, config.amb, config.ann, config.bwt, config.pac, config.sa]
+    if getattr(config, 'r2', None):  # If a fastq pair was provided
+        file_names.insert(1, 'r2.fq.gz')
+        ids.insert(1, config.r2)
+    if getattr(config, 'alt', None):  # If an alt file was provided
+        file_names.append('ref.fa.alt')
+        ids.append(config.alt)
+    for fileStoreID, name in zip(ids, file_names):
+        job.fileStore.readGlobalFile(fileStoreID, os.path.join(work_dir, name))
+    # Read groups
```

Comment:
This is not a useful comment. I'd go with something more like "Was a read group line provided? If not, generate this line by using the UUID blah blah blah"